seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-data-structure-questions.json

```diff
@@ -203,35 +203,54 @@
-        "assert((function(){ var test = new CircularQueue(); test.enqueue('a'); test.enqueue('b'); return test.front() === 'a' && test.front() === 'a'; })(), 'message: Your <code>front</code> method should return the item at the current read pointer position, without removing it.');"
+        "console.assert((function(){ var test = new CircularQueue(3); test.enqueue(17); test.enqueue(32); test.enqueue(591); var print = test.print(); return print[0] === 17 && print[1] === 32 && print[2] === 591; })(), 'message: The <code>enqueue</code> method adds items to the circular queue.');",
+        "console.assert((function(){ var test = new CircularQueue(3); test.enqueue(17); test.enqueue(32); test.enqueue(591); test.enqueue(13); test.enqueue(25); test.enqueue(59); var print = test.print(); return print[0] === 17 && print[1] === 32 && print[2] === 591; })(), 'message: You cannot enqueue items past the read pointer.');",
+        "console.assert((function(){ var test = new CircularQueue(3); test.enqueue(17); test.enqueue(32); test.enqueue(591); return test.dequeue() === 17 && test.dequeue() === 32 && test.dequeue() === 591; })(), 'message: The <code>dequeue</code> method dequeues items from the queue.');",
+        "console.assert((function(){ var test = new CircularQueue(3); test.enqueue(17); test.enqueue(32); test.enqueue(672); test.dequeue(); test.dequeue(); var print = test.print(); return print[0] === null && print[1] === null && print[2] === 672; })(), 'message: After an item is dequeued its position in the queue should be reset to <code>null</code>.');",
+        "console.assert((function(){ var test = new CircularQueue(3); test.enqueue(17); test.enqueue(32); test.enqueue(591); return test.dequeue() === 17 && test.dequeue() === 32 && test.dequeue() === 591 && test.dequeue() === null && test.dequeue() === null && test.dequeue() === null && test.dequeue() === null && test.enqueue(100) === 100 && test.dequeue() === 100; })(), 'message: Trying to dequeue past the write pointer returns <code>null</code> and does not advance the write pointer.');"
       ],
       "type": "waypoint",
       "releasedOn": "Feb 17, 2017",
-      "solutions": [
-        "class CircularQueue {\n  constructor () {\n    this._cq = ['', '', '', '', ''];\n    this._reader = 0;\n    this._writer = 0;\n  }\n  \n  printCollection () {\n    return this._cq;\n  }\n  \n  _updatePointer (id) {\n    if (++this[id] === this._cq.length) {\n      this[id] = 0;\n    }\n  }\n  \n  enqueue (item) {\n    if (Array.isArray(item)) {\n      for (const x of item) {\n        this.enqueue(x);\n      }\n\n      return;\n    }\n    \n    this._cq[this._writer] = item;\n    \n    this._updatePointer('_writer');\n  }\n  \n  dequeue () {\n    const item = this._cq[this._reader];\n    this._cq[this._reader] = '';\n    \n    this._updatePointer('_reader');\n    return item;\n  }\n  \n  front () {\n    return this._cq[this._reader];\n  }\n}"
```

Comment:
Ah, I created this solution for the old version of the challenge, so it's no longer valuable.

I think we should add an updated solution.